odo/backends/sql.py

```diff
@@ -199,6 +220,8 @@ def discover_typeengine(typ):
                              'second_precision=%d, day_precision=%d' %
                              (typ.second_precision, typ.day_precision))
         return datashape.TimeDelta(unit=units)
+    if type(typ) in precision_types and typ.precision in precision_to_dtype:
```

Comment:
what happens when the first is true but the second is not?

Reply:
A `sa.Float(precision=42)` (where 42 is not a mapped precision value) would be treated as a `sa.Float` by the `if type(typ) in revtypes` predicate.

A postgres `DOUBLE_PRECISION` would return a `Decimal(precision=42)`, having gone through the logic which this patch circumvents when the precision is 53.

I'm not sure what should be the correct logic when there is a precision but it is not 24 or 53.
A couple options:
1) If the precision is not None is not mapped, raise an exception (Something like `UnexpectedColumnPrecision`.)  If the precision is None, then use the mapping in `revtypes`.
e.g. `sa.Float(precision=42)` raises an exception; `sa.Float()` returns `float64` (The same dtype as `sa.Float`)
2) If the precision is not mapped, then just continue on with the previously defined dtype mapping logic . (This is the logic the PR currently implements.)
In this case, `sa.Float(precision=8)`, `sa.Float(precision=42)`, `sa.Float()` all return `float64`.

I lean towards raising the exception on unrecognized precisions.

Reply:
I agree, raising an exception seems like the best thing to do for now.

Reply:
Actually, what is wrong with returning 64? It will be able to hold the value.

Reply:
How about instead of mapping just 24 and 53, instead mapping based on ranges of values:
- `1-24` to `float32`
- `25-53` to `float64`
- `>=54` raises an exception.

This would follow the behavior of Postgres.
From https://www.postgresql.org/docs/9.5/static/datatype-numeric.html#DATATYPE-NUMERIC-DECIMAL,
```
PostgreSQL also supports the SQL-standard notations float and float(p) for specifying inexact numeric types. Here, p specifies the minimum acceptable precision in binary digits. PostgreSQL accepts float(1) to float(24) as selecting the real type, while float(25) to float(53) select double precision. Values of p outside the allowed range draw an error. float with no precision specified is taken to mean double precision
```

Reply:
Thinking about the `>=54`, since that is not a valid value on the database side, that would not be a precision returned during reflection, so just the supported ranges would need to be handled.

Reply:
I think it doesn't hurt to guard against that for now, I like the range mappings.

Reply:
https://github.com/blaze/odo/pull/592/commits/a3785c6aaf8dcd98237bca27e5ff1079a910f480